packages/cli/src/utils/modify_config.ts

```diff
@@ -0,0 +1,193 @@
+
+        await fs.promises.writeFile(filePath, newCode, {
+          encoding: 'utf-8',
+          flag: 'w',
+        })
+
+        highlightCodeDifferences(oldCode, newCode, detectedBuildTool)
+      }
+    } else if (detectedModuleType === 'esm') {
+      // 1.
+      const importStatement = `import million from 'million/compiler'\n`
+      configFileContent = importStatement + configFileContent
+
+      // 2.
+      const regex = /export\s+default\s+([^;]+)/
+      const match = configFileContent.match(regex)
+
+      if (match) {
+        const oldExportExpression = match[1]
+        const newExportExpression = await wrapExportCode(detectedBuildTool, oldExportExpression!)
+
+        // 3.
+        const newCode = configFileContent.replace(regex, `export default ${newExportExpression}`)
+
+        await fs.promises.writeFile(filePath, newCode, {
+          encoding: 'utf-8',
+          flag: 'w',
+        })
+
+        highlightCodeDifferences(oldCode, newCode, detectedBuildTool)
+      }
+    } else {
+      /**
+       *  If we can't detect the module type, we can't modify the config file
+       *  Refer user to read installation docs
+       * */
+
+      abort(
+        chalk.yellow(`Could not detect module type for ${detectedBuildTool.configFilePath}.`) +
+          `\nPlease refer docs:` +
+          chalk.cyan('https://million.dev/docs/install#use-the-compiler') +
+          ` to setup manually.`,
+      )
+    }
+  } catch (err) {
+    console.log(err)
+    return abort()
+  }
+}
+
+export async function getExistingConfigContent(detectedBuildTool: BuildTool): Promise<string> {
+  const configFilePath = path.join(process.cwd(), detectedBuildTool.configFilePath)
+
+  // Read the config file
+  const configContent = await fs.promises.readFile(configFilePath, { encoding: 'utf-8' })
+  return configContent
+}
+
+function detectModuleType(fileContent: string): 'cjs' | 'esm' | 'unknown' {
+  // Check for CommonJS require/module.exports patterns
+  if (fileContent.includes('module.exports =')) {
+    return 'cjs'
+  }
+
+  // Check for ESM import/export statements
+  if (fileContent.includes('export default')) {
+    return 'esm'
+  }
+
+  return 'unknown'
+}
+
+async function wrapExportCode(buildtool: BuildTool, oldExportExpression: string): Promise<string> {
+  let [firstPart, ...rest]: string[] = []
+  let newExportExpression: string = ''
+
+  switch (buildtool.name) {
+    case 'next':
+      /**
+       * million.next(nextConfig, millionConfig);
+       * */
+      const nextRouter: 'app' | 'pages' = await getNextRouter()
+      return (newExportExpression = `million.${buildtool.bundler}(
+  ${oldExportExpression}, { auto: ${nextRouter == 'app' ? '{ rsc: true }' : 'true'}}
+)`)
+    case 'vite':
+      /**
+       * defineConfig({
+       *   plugins: [million.vite({ auto: true }), react(), ],
+       * });
+       */
+      ;[firstPart, ...rest] = oldExportExpression.split('plugins: [')
```

Comment:
why can't there be a const here

Reply:
Initialised [firstpart, ...rest] before the switch case.

Initialising for each case was causing re-declaration warning.